batchlib/analysis/cell_level_analysis.py

```diff
@@ -283,24 +288,78 @@ def save_all_stats(self, in_file, out_file,
                 self.write_table(f, output_key, columns, table)
 
     def get_bg_stats(self, bg_values):
+        if bg_values is None:
+            return {'median': [np.nan] * len(self.channel_keys),
+                    'mad': [np.nan] * len(self.channel_keys)}
         # bg_vales should have shape n_channels, n_pixels
         bg_values = bg_values.cpu().numpy().astype(np.float32)
         medians = np.median(bg_values, axis=1)
         mads = np.median(np.abs(bg_values - medians[:, None]), axis=1)
 
         return {'median': medians, 'mad': mads}
 
+    # TODO the three functions below are copied from CellAnalysis.
+    #  Solve this better.. Maybe put them in a mixin class?
+    @staticmethod
+    def folder_to_table_path(folder, identifier):
+        # NOTE, we call this .hdf5 to avoid pattern matching, it's a bit hacky ...
+        table_file_name = os.path.split(folder)[1] + '_table.hdf5'
+        return os.path.join(folder, table_file_name)
+
+    @property
+    def table_out_path(self):
+        return self.folder_to_table_path(self.folder, self.identifier)
+
+    def load_image_outliers(self, input_files):
```

Comment:
Refactor this into `load_image_outlier_dict` and call it in the methods.

Reply:
This is addressed now. You can have a look if you want. However I did not figure out the job order / input validation of the image qc job yet.